crates/turborepo/src/main.rs

```diff
@@ -43,14 +43,23 @@ fn run_go_binary(args: Args) -> Result<i32> {
 
     let serialized_args = serde_json::to_string(&args)?;
     trace!("Invoking go binary with {}", serialized_args);
-    let mut command = process::Command::new(go_binary_path)
+    let mut command = process::Command::new(go_binary_path);
+    command
         .arg(serialized_args)
         .stdout(Stdio::inherit())
-        .stderr(Stdio::inherit())
-        .spawn()
-        .expect("Failed to execute turbo.");
+        .stderr(Stdio::inherit());
 
-    let exit_code = command.wait()?.code().unwrap_or(2);
+    let shared_child = shared_child::SharedChild::spawn(&mut command).unwrap();
+    let child_arc = std::sync::Arc::new(shared_child);
+
+    let child_arc_clone = child_arc.clone();
```

Comment:
When we use `child_arg_clone` in the closure below, it gets "moved" into the closure, basically the closure owns the variable and it cannot be used anywhere else. If we were to just use `child_arg` in the closure, this line:

```
    let exit_code = child_arc.wait()?.code().unwrap_or(2);
```

would not work because `child_arg` would have been moved into the closure.

Reply:
gotcha. this is a Rust concept of "ownership" then? Thanks for the explanation!

Reply:
Basically there is no runtime magic to keep track of references so cloning an arc increments the reference counter :)

https://doc.rust-lang.org/src/alloc/sync.rs.html#76-81